xds/internal/resolver/xds_resolver.go

```diff
@@ -81,6 +90,10 @@ func (b *xdsResolverBuilder) Build(t resolver.Target, cc resolver.ClientConn, op
 		return nil, fmt.Errorf("xds: failed to create xds-client: %v", err)
 	}
 	r.client = client
+	bootstrapConfig := client.BootstrapConfig()
```

Comment:
This is probably a bug that we did bootstrap only when xds creds were configured. Should this be release noted?

Reply:
And the same for the case where `Build()` fails, and we might be leaking the xdsClient. This was was possible only on the old line 99. Not sure if we need to release note that.